python/tests/test_metrics.py

```diff
@@ -82,3 +84,47 @@ def test_multi_class_metrics(
 
         evaluator.setMetricName(metric_name)  # type: ignore
         assert math.fabs(evaluator.evaluate(sdf) - metrics.evaluate((evaluator))) < 1e-6
+
+
+def get_regression_metrics(
+    pdf: pd.DataFrame, label_name: str, prediction_name: str
+) -> RegressionMetrics:
+    pdf = pdf.copy(True)
+    pdf.insert(1, "gap", pdf[label_name] - pdf[prediction_name])
+    mean = pdf.mean()
+    m2 = pdf.pow(2).sum()
+    l1 = pdf.abs().sum()
+    total_cnt = pdf.shape[0]
+    m2n = pdf.var(ddof=0) * pdf.shape[0]
+
+    return RegressionMetrics.create(mean, m2n, m2, l1, total_cnt)
+
+
+@pytest.mark.parametrize("metric_name", ["rmse", "mse", "r2", "mae", "var"])
+def test_regression_metrics(metric_name: str) -> None:
+    columns = ["label", "prediction"]
+    np.random.seed(10)
+    pdf1 = pd.DataFrame(
+        np.random.uniform(low=-20, high=20, size=(1000, 2)), columns=columns
+    ).astype(np.float64)
+    np.random.seed(100)
+    pdf2 = pd.DataFrame(
+        np.random.uniform(low=-20, high=20, size=(1000, 2)), columns=columns
```

Comment:
Can you make this size different from pdf1?   Say 2000?

Reply:
Good suggestion, I also added the max_record_batch parameter to test multiple batch scenario.